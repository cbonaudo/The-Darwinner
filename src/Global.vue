<template>
  <div>
    <Goal></Goal>
    <Header></Header>
    <Cells></Cells>
    <Changelog></Changelog>
  </div>
</template>

<script>
import Goal from "./components/Goal";
import Header from "./components/Header";
import Cells from "./components/Cells";
import Changelog from "./components/Changelog";

export default {
  name: "Global",
  components: { Goal, Header, Cells, Changelog }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1,
h2 {
  font-weight: normal;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
