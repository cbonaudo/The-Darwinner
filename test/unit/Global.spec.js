import Global from '@/Global'
import { mount } from '@vue/test-utils'

describe('Global.vue', () => {
  const wrapper = mount(Global)

  it('should render correct contents', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have correct starting data', () => {
    expect(Global.data()).toMatchSnapshot()
  })

  describe('resetAll', () => {
    it('should reset the clicks', () => {
      wrapper.setData({ clicks: 10 })
      expect(wrapper.vm.clicks).toEqual(10)
      wrapper.vm.resetAll()
      expect(wrapper.vm.clicks).toEqual(0)
    })
    it('should reset the keyStrokes', () => {
      wrapper.setData({ keyStrokes: 10 })
      expect(wrapper.vm.keyStrokes).toEqual(10)
      wrapper.vm.resetAll()
      expect(wrapper.vm.keyStrokes).toEqual(0)
    })
  })

  describe('unlockKeystroke', () => {
    it('should add 65 to keyStrokesAllowed', () => {
      wrapper.setData({ keyStrokesAllowed: [] })
      expect(wrapper.vm.keyStrokesAllowed).toEqual([])
      wrapper.vm.unlockKeystroke(65)
      expect(wrapper.vm.keyStrokesAllowed).toEqual([65])
    })
    it('should add 120 to keyStrokesAllowed with 65', () => {
      wrapper.setData({ keyStrokesAllowed: [65] })
      expect(wrapper.vm.keyStrokesAllowed).toEqual([65])
      wrapper.vm.unlockKeystroke(120)
      expect(wrapper.vm.keyStrokesAllowed).toEqual([65, 120])
    })
  })
})
