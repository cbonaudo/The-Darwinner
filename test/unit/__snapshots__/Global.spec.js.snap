// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global.vue should have correct starting data 1`] = `
Object {
  "clicks": 0,
  "era": 1,
  "keyStrokes": 0,
  "keyStrokesAllowed": Array [],
}
`;

exports[`Global.vue should render correct contents 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
